nixos/modules/programs/skim.nix

```diff
@@ -0,0 +1,26 @@
+{ pkgs, config, lib, ... }:
+with lib;
+let
+  cfg = config.programs.skim;
+in
+{
+  options = {
+    programs.skim = {
+      fuzzyCompletion = mkEnableOption (mdDoc "fuzzy Completion with skim");
+      keybindings = mkEnableOption (mdDoc "skim keybindings");
+    };
+  };
+  config = {
+    environment.systemPackages = optional (cfg.keybindings || cfg.fuzzyCompletion) pkgs.skim;
```

Comment:
Don't pass `pkgs.skim` directly. Otherwise, the end user will not be able to override it.

Instead, use `cfg.package` with accompaning `option.programs.skim.package`.